File: GO/ex5.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)

	fmt.Print("Digite uma frase: ")
	frase, _ := reader.ReadString('\n')
	frase = strings.TrimSpace(frase)

	fmt.Print("Digite a palavra a ser contada: ")
	palavra, _ := reader.ReadString('\n')
	palavra = strings.TrimSpace(palavra)

	frase = strings.ToUpper(frase)
	palavra = strings.ToUpper(palavra)

	contagem := 0
	for i := 0; i <= len(frase)-len(palavra); i++ {
		if frase[i:i+len(palavra)] == palavra {
			contagem++
		}
	}

	fmt.Printf("A palavra '%s' ocorre %d vezes na frase.\n", palavra, contagem)
}

File: Haskell Script/ex5.hs
import Data.List
import Data.Char (toUpper)

main :: IO ()
main = do
    putStrLn "Digite uma frase: "
    frase <- getLine
    putStrLn "Digite a palavra a ser contada: "
    palavra <- getLine
    let fraseUpper = map toUpper frase
    let palavraUpper = map toUpper palavra
    let contagem = contarOcorrencias palavraUpper fraseUpper
    putStrLn $ "A palavra '" ++ palavra ++ "' ocorre " ++ show contagem ++ " vezes na frase."

contarOcorrencias :: String -> String -> Int
contarOcorrencias palavra = length . filter (isPrefixOf palavra) . tails
